packages/@lwc/ssr-compiler/src/compile-template/ir-to-es.ts

```diff
@@ -66,14 +66,23 @@ const transformers: Transformers = {
     Lwc: LwcComponent,
 };
 
-export function irChildrenToEs(children: IrChildNode[], cxt: TransformerContext): EsStatement[] {
-    const result = children.flatMap((child, idx) => {
-        cxt.prevSibling = children[idx - 1];
-        cxt.nextSibling = children[idx + 1];
-        return irToEs(child, cxt);
-    });
+export function irChildrenToEs(
+    children: IrChildNode[],
+    cxt: TransformerContext,
+    cb?: (child: IrChildNode) => void
+): EsStatement[] {
+    const result: EsStatement[] = [];
+
+    for (let i = 0; i < children.length; i++) {
+        cxt.prevSibling = children[i - 1];
+        cxt.nextSibling = children[i + 1];
+        cb?.(children[i]);
+        result.push(...irToEs(children[i], cxt));
+    }
+
     cxt.prevSibling = undefined;
     cxt.nextSibling = undefined;
+
```

Comment:
I added a callback to allow the `prevSibling` and `nextSibling` logic to be reused, and also changed the flat map to a for loop as it should be easier to reason about and debug. I'm a huge flatMap fan, but using it here doesn't seem right. That said, I can keep the change to just the callback.